--- pearpc-0.4/src/tools/crc32.cc~	2006-01-05 14:10:28.000000000 +0100
+++ pearpc-0.4/src/tools/crc32.cc	2006-01-05 14:07:53.000000000 +0100
@@ -106,7 +106,7 @@
     uint32  crc = 0xffffffff;   // preload shift register, per CRC-32 spec
 
 #ifdef RUN_FASTER
-    for (; (((uint32)p)&0x03)!=0 && len>0; len--)
+    for (; (((uintptr_t)p)&0x03)!=0 && len>0; len--)
         DO_CRC(*p++);           // will execute if *p is not dword aligned
     for (; len>=sizeof(uint32); p += sizeof(uint32), len -= sizeof(uint32)) {
         uint32 data = *(uint32*)p;
